File: 1310-xor-queries-of-a-subarray/1310-xor-queries-of-a-subarray.java
class Solution {
    public int[] xorQueries(int[] arr, int[][] queries) {
        int n = arr.length;
        for (int i = 1; i < n; i++) {
            arr[i] ^= arr[i-1];
        }
        int m = queries.length;
        int[] ans = new int[m];
        for (int i = 0; i < m; i++) {
            int l = queries[i][0], r = queries[i][1];
            ans[i] = arr[r] ^ (l == 0 ? 0 : arr[l-1]);
        }
        return ans;
    }
}

File: LeetCode/1310-xor-queries-of-a-subarray/1310-xor-queries-of-a-subarray.py
class Solution:
    def xorQueries(self, arr: List[int], queries: List[List[int]]) -> List[int]:
        n = len(arr)
        for i in range(1, n):
            arr[i] ^= arr[i-1]
        m = len(queries)
        ans = [0] * m
        for i in range(m):
            l, r = queries[i][0], queries[i][1]
            ans[i] = arr[r] ^ (0 if l == 0 else arr[l-1])
        return ans